src/librustc_lint/unused.rs

```diff
@@ -425,6 +425,7 @@ trait UnusedDelimLint {
                     ExprKind::Ret(_) | ExprKind::Break(..) => true,
                     _ => parser::contains_exterior_struct_lit(&inner),
                 })
+            || if let ExprKind::Yield(..) = inner.kind { true } else { false }
```

Comment:
Isn't this slightly overeager?

From what I can tell this does not warn for `foo((yield))`.

I think it should be fine to move `ExprKind::Yield` into the inner match :thinking:

Reply:
My concern was that it emits duplicate warnings (in `BlockRetValue`?) and I haven't figured out ideal solution... But yeah it's better to warn needed cases than to reduce warnings in edge cases.
Addressed via 2e7ba78 and left some FIXME comments in the test.